lib/query.js

```diff
@@ -320,42 +289,28 @@ Query.prototype.select = function (bins) {
  * })
  *
  * @see {@link module:aerospike/filter} to create secondary index filters.
- * @see {@link module:aerospike/predexp} to create predicate expression filters.
  *
  * @example <caption>Applying a predicate expression in addition to a secondary
  * index filter to find only record where the 'tags' list bin contains the
  * value 'blue' but also contains at least one other value that is not
  * 'blue':</caption>
  *
  * const Aerospike = require('aerospike')
- * const predexp = Aerospike.predexp
  *
  * Aerospike.connect().then(client => {
  *   let query = client.query('test', 'demo')
  *
  *   let tagsFilter = Aerospike.filter.contains('tags', 'blue', Aerospike.indexType.LIST)
  *   query.where(tagsFilter)
  *
- *   let tagsPredexp = [
- *     predexp.stringVar('tag'),
- *     predexp.stringValue('blue'),
- *     predexp.stringEqual(),
- *     predexp.listBin('tags'),
- *     predexp.listIterateAnd('tag'),
- *     predexp.not()
- *   ]
- *   query.where(tagsPredexp)
- *
  *   let stream = query.foreach()
  *   stream.on('data', record => { console.info(record.bins.tags) })
  *   stream.on('error', error => { throw error })
  *   stream.on('end', () => client.close())
  * })
  */
 Query.prototype.where = function (predicate) {
-  if (Array.isArray(predicate) && predicate[0] instanceof predexp.PredicateExpression) {
-    this.setPredexpFilter(predicate)
-  } else if (predicate instanceof filter.SindexFilterPredicate) {
+  if (predicate instanceof filter.SindexFilterPredicate) {
```

Comment:
I think the client should raise a TypeError if the predicate is not a filter predicate.